src/app/portlets/dot-edit-page/content/services/html/dot-edit-content-toolbar-html.service.spec.ts

```diff
@@ -142,31 +142,64 @@ describe('DotEditContentToolbarHtmlService', () => {
     });
 
     describe('contentlet toolbar', () => {
+        let htmlElement: HTMLHtmlElement;
+
         beforeEach(() => {
             testDoc = document.implementation.createDocument('http://www.w3.org/1999/xhtml', 'html', null);
             dummyContainer = testDoc.createElement('div');
-            const htmlElement: HTMLHtmlElement = testDoc.getElementsByTagName('html')[0];
-            dummyContainer.innerHTML = `
-                <div data-dot-object="container">
-                    <div data-dot-object="contentlet" data-dot-can-edit="false">
-                        <div class="large-column"></div>
-                    </div>
-                </div>
-            `;
-            htmlElement.appendChild(dummyContainer);
-            dotEditContentToolbarHtmlService.addContentletMarkup(testDoc);
+            htmlElement = testDoc.getElementsByTagName('html')[0];
         });
 
-        it('should create buttons', () => {
-            expect(testDoc.querySelectorAll('.dotedit-contentlet__drag').length).toEqual(1);
-            expect(testDoc.querySelectorAll('.dotedit-contentlet__edit').length).toEqual(1);
-            expect(testDoc.querySelectorAll('.dotedit-contentlet__remove').length).toEqual(1);
-        });
+        describe('default', () => {
+            beforeEach(() => {
+                dummyContainer.innerHTML = `
+                    <div data-dot-object="container">
+                        <div data-dot-object="contentlet" data-dot-can-edit="false">
+                            <div class="large-column"></div>
+                        </div>
+                    </div>
+                `;
+                htmlElement.appendChild(dummyContainer);
+                dotEditContentToolbarHtmlService.addContentletMarkup(testDoc);
+            });
+
+            it('should create buttons', () => {
+                expect(testDoc.querySelectorAll('.dotedit-contentlet__drag').length).toEqual(1);
+                expect(testDoc.querySelectorAll('.dotedit-contentlet__edit').length).toEqual(1);
+                expect(testDoc.querySelectorAll('.dotedit-contentlet__remove').length).toEqual(1);
+                expect(testDoc.querySelectorAll('.dotedit-contentlet__code').length).toEqual(0);
+            });
+
+            it('should have edit button disabled', () => {
+                expect(testDoc.querySelector('.dotedit-contentlet__edit').classList.contains('dotedit-contentlet__disabled')).toBe(true);
+            });
 
-        it('should have edit button disabled', () => {
-            expect(testDoc.querySelector('.dotedit-contentlet__edit').classList.contains('dotedit-contentlet__disabled')).toBe(true);
+            xit('should bind events');
```

Comment:
xit???

Reply:
Yup... this is old, it means we have to test that but the ways is right now we can't